HIRS_Utils/src/main/java/hirs/data/persist/certificate/Certificate.java

```diff
@@ -328,6 +331,47 @@ public Certificate(final byte[] certificateBytes) throws IOException {
         this.certAndTypeHash = Objects.hash(certificateHash, getClass().getSimpleName());
     }
 
+    @SuppressWarnings("magicnumber")
+    private byte[] trimCertificate(final byte[] certificateBytes) {
+        int certificateStart = 0;
+        int certificateLength = 0;
```

Comment:
This method would be less cryptic and cleaner if you used a `ByteBuffer`.

Reply:
I'm not sure how this is less cryptic or cleaner than the one that's here. At least with the current version, the for loop is straightforward and you don't have to ponder where you're at in the ByteBuffer compared to simple array indexing.

Also, I ran the tests and they passed, so I'm assuming that this non-pushed version was correctly implemented.

![image](https://user-images.githubusercontent.com/36857962/47040866-5f886600-d155-11e8-9e16-93594a85600e.png)

Reply:
Went ahead and pushed the work for shifting to a ByteBuffer. Let me know what you think.